gr-iio/lib/attr_source_impl.cc

```diff
@@ -0,0 +1,285 @@
+{
+    switch (attr_type) {
+    case 0:
+        ret = iio_channel_attr_read_double(chan, attribute.c_str(), value);
+        break;
+    case 1:
+        ret = iio_device_attr_read_double(dev, attribute.c_str(), value);
+        break;
+    default:
+        ret = iio_device_debug_attr_read_double(dev, attribute.c_str(), value);
+        break;
+    }
+    attr_source_impl::check(ret);
+}
+
+void attr_source_impl::get_attribute_data(const std::string& attribute, float* value)
+{
+    double dvalue;
+    switch (attr_type) {
+    case 0:
+        ret = iio_channel_attr_read_double(chan, attribute.c_str(), &dvalue);
+        break;
+    case 1:
+        ret = iio_device_attr_read_double(dev, attribute.c_str(), &dvalue);
+        break;
+    default:
+        ret = iio_device_debug_attr_read_double(dev, attribute.c_str(), &dvalue);
+        break;
+    }
+    attr_source_impl::check(ret);
+    *value = boost::lexical_cast<float>(dvalue);
+}
+
+void attr_source_impl::get_attribute_data(const std::string& attribute, long long* value)
+{
+    switch (attr_type) {
+    case 0:
+        ret = iio_channel_attr_read_longlong(chan, attribute.c_str(), value);
+        break;
+    case 1:
+        ret = iio_device_attr_read_longlong(dev, attribute.c_str(), value);
+        break;
+    default:
+        ret = iio_device_debug_attr_read_longlong(dev, attribute.c_str(), value);
+        break;
+    }
+    attr_source_impl::check(ret);
+}
+
+void attr_source_impl::get_attribute_data(const std::string& attribute, int* value)
+{
+    long long llvalue;
+    switch (attr_type) {
+    case 0:
+        ret = iio_channel_attr_read_longlong(chan, attribute.c_str(), &llvalue);
+        break;
+    case 1:
+        ret = iio_device_attr_read_longlong(dev, attribute.c_str(), &llvalue);
+        break;
+    default:
+        ret = iio_device_debug_attr_read_longlong(dev, attribute.c_str(), &llvalue);
+        break;
+    }
+    attr_source_impl::check(ret);
+    *value = boost::lexical_cast<int>(llvalue);
+}
+
+void attr_source_impl::get_attribute_data(const std::string& attribute, uint8_t* value)
+{
+    bool bvalue;
+    switch (attr_type) {
+    case 0:
+        ret = iio_channel_attr_read_bool(chan, attribute.c_str(), &bvalue);
+        break;
+    case 1:
+        ret = iio_device_attr_read_bool(dev, attribute.c_str(), &bvalue);
+        break;
+    default:
+        ret = iio_device_debug_attr_read_bool(dev, attribute.c_str(), &bvalue);
+        break;
+    }
+    attr_source_impl::check(ret);
+    *value = boost::lexical_cast<uint8_t>(bvalue);
+}
+
+int attr_source_impl::work(int noutput_items,
+                           gr_vector_const_void_star& input_items,
+                           gr_vector_void_star& output_items)
+{
+    int sample;
+    void* out;
+
+    out = output_items[0];
+
+    for (sample = 0; sample < samples_per_update; sample++) {
+        boost::this_thread::sleep_for(boost::chrono::milliseconds(update_interval_ms));
```

Comment:
GR has moved to std::chrono, consider switching also to remove the dependency on boost::chrono. Suggest switching the includes to `<chrono>` and `<thread>` and using:

```c++
std::this_thread::sleep_for(std::chrono::milliseconds(update_interval_ms));
```